src/reformatters/noaa/gefs/gefs_config_models.py

```diff
@@ -20,3 +25,218 @@ class GEFSInternalAttrs(NoaaInternalAttrs):
+            resolution_str = FILE_RESOLUTIONS[true_gefs_file_type]
+            url = (
+                "https://noaa-gefs-pds.s3.amazonaws.com/"
+                f"gefs.{init_date_str}/{init_hour_str}/atmos/pgrb2{true_gefs_file_type}{resolution_str.strip('0')}/"
+                f"ge{ensemble_or_statistic_str}.t{init_hour_str}z.pgrb2{true_gefs_file_type}.{resolution_str}.f{lead_time_hours:03.0f}"
+            )
+            return url
+        elif self.init_time >= GEFS_REFORECAST_END:
+            url = (
+                "https://noaa-gefs-pds.s3.amazonaws.com/"
+                f"gefs.{init_date_str}/{init_hour_str}/pgrb2{true_gefs_file_type}/"
+                f"ge{ensemble_or_statistic_str}.t{init_hour_str}z.pgrb2{true_gefs_file_type}f{lead_time_hours:02.0f}"
+            )
+            return url
+        else:
+            assert len(data_vars) == 1, "Only one data variable per file in GEFS v12 reforecast"  # fmt: skip
+            data_var = data_vars[0]
+            days_str = (
+                "Days:1-10"
+                if self.lead_time <= pd.Timedelta(hours=240)
+                else "Days:10-16"
+            )
+            grib_element = get_grib_element(data_var, self.init_time)
+            level_str = GEFS_REFORECAST_LEVELS_SHORT[data_var.internal_attrs.grib_index_level]  # fmt: skip
+            url = (
+                "https://noaa-gefs-retrospective.s3.amazonaws.com/"
+                f"GEFSv12/reforecast/{self.init_time.year}/{init_date_str}{init_hour_str}/"
+                f"{ensemble_or_statistic_str}/{days_str}/"
+                f"{grib_element.lower()}_{level_str}_"
+                f"{init_date_str}{init_hour_str}_{ensemble_or_statistic_str}.grib2"
+            )
+
+            return url
+
+
+class GefsEnsembleSourceFileCoord(GefsSourceFileCoord):
+    """Source file coordinate for GEFS forecast ensemble data."""
+
+    ensemble_member: int
+
+    def out_loc(self) -> Mapping[Dim, CoordinateValueOrRange]:
+        """Use existing coordinate structure directly for forecast dataset."""
+        return {
+            "init_time": self.init_time,
+            "lead_time": self.lead_time,
+            "ensemble_member": self.ensemble_member,
+        }
+
+
+class GefsStatisticSourceFileCoord(GefsSourceFileCoord):
+    """Source file coordinate for GEFS forecast statistic data."""
+
+    statistic: EnsembleStatistic
+
+    def out_loc(self) -> Mapping[Dim, CoordinateValueOrRange]:
+        """Use existing coordinate structure directly for forecast dataset.
+
+        Note: Statistic variables in forecast dataset don't have a 'statistic' dimension.
+        They are separate data variables that don't include the ensemble_member dimension.
+        The statistic coordinate is used for file selection during download/read,
+        but the output location only uses init_time and lead_time.
+        """
+        return {
+            "init_time": self.init_time,
+            "lead_time": self.lead_time,
+        }
```

Comment:
We aren't technically using this yet but we'll have a statistics dimension to write into
```suggestion
        return {
            "init_time": self.init_time,
            "lead_time": self.lead_time,
            "statistic": self.statistic,
        }
```

Reply:
But `statistic` isn't on ` reformatters.common.types.Dim`, do we want that everywhere?